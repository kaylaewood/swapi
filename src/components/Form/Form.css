form {
  background-color: #0B1E36;
  color: white;
  padding: 1vw 2vw 2vw 2vw;
  width: 45%;
  margin: auto;
  margin-top: 2vw;
  letter-spacing: .1vw;
  border-radius: 2%;
  box-shadow: 15px 15px 15px #A6A6A6;
}

.form-text {
  padding-top: 1vw;
}

input,
select,
button {
  width: 98%;
  font-size: 1.2vw;
  height: 3vw;
}

button {
  background-color: #414649;
  color: white;
  font-weight: 600;
  font-size: 2vw;
  border: none;
  letter-spacing: .3vw;
  margin-top: 2vw;
}

button:hover,
select:hover,
button:focus,
select:focus {
  transform: scale(1.05);
  cursor: pointer;
}

input:hover,
input:focus {
  transform: scale(1.05);
}

.show {
  color: red;
}

.hide {
  display: none;
}
