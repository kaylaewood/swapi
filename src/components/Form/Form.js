import React, { Component } from 'react';
import './Form.css';
import { Link } from 'react-router-dom';

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      favoriteQuote: '',
      ranking: 'jedi-master',
      formComplete: 'hide'
    }
  }

  updateUserInfo = event => {
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  handleClick = () => {
    if(this.state.name && this.state.favoriteQuote && this.state.ranking) {
      this.setState({ formComplete: 'hide' });
      this.props.updateStateFromForm(this.state.name, this.state.favoriteQuote, this.state.ranking);
    } else {
      this.setState({ formComplete: 'show' })
    }
  }

  render() {
    return (
      <form>
        <p className='form-text'>WHAT IS YOUR FIRST NAME?</p>
        <input
          type='text'
          name='name'
          value={this.state.name}
          onChange={this.updateUserInfo}
        />
        <p className='form-text'>WHAT IS YOUR FAVORITE STAR WARS QUOTE?</p>
        <input
          type='text'
          name='favoriteQuote'
          value={this.state.favoriteQuote}
          onChange={this.updateUserInfo}
        />
        <p className='form-text'>HOW WELL DO YOU KNOW STAR WARS?</p>
        <select
          name='ranking'
          value={this.state.ranking}
          onChange={this.updateUserInfo}
        >
          <option value='jedi-master'>JEDI MASTER</option>
          <option value='jedi-knight'>JEDI KNIGHT</option>
          <option value='padawan'>PADAWAN</option>
        </select>
        <span className={this.state.formComplete.toString()}>ALL FIELDS MUST BE FILLED IN</span>
        <Link to='/movies' className='login-button'>
          <button type='button' onClick={this.handleClick}>LET'S GO</button>
        </Link>
      </form>
    )
  }
};
export default Form;
