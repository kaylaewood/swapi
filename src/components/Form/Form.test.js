import React from 'react';
import ReactDOM from 'react-dom';
import Form from './Form';
import { shallow } from 'enzyme';

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Form />)
    expect(wrapper).toMatchSnapshot();
    })
})
