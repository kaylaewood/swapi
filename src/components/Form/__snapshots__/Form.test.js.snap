// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with the favorited class 1`] = `
<form>
  <p
    className="form-text"
  >
    WHAT IS YOUR FIRST NAME?
  </p>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <p
    className="form-text"
  >
    WHAT IS YOUR FAVORITE STAR WARS QUOTE?
  </p>
  <input
    name="favoriteQuote"
    onChange={[Function]}
    type="text"
    value=""
  />
  <p
    className="form-text"
  >
    HOW WELL DO YOU KNOW STAR WARS?
  </p>
  <select
    name="ranking"
    onChange={[Function]}
    value="jedi-master"
  >
    <option
      value="jedi-master"
    >
      JEDI MASTER
    </option>
    <option
      value="jedi-knight"
    >
      JEDI KNIGHT
    </option>
    <option
      value="padawan"
    >
      PADAWAN
    </option>
  </select>
  <span
    className="hide"
  >
    ALL FIELDS MUST BE FILLED IN
  </span>
  <button
    onClick={[Function]}
    type="button"
  >
    LET'S GO
  </button>
</form>
`;
